app/helpers/application_helper.rb

```diff
@@ -1,7 +1,11 @@
 module ApplicationHelper
+  def self.find(id)
+    User.find_by(id: id) || NullUser.new
+  end
+
   def current_user
     if session[:user_id]
-      @current_user ||= User.find_by(id: session[:user_id])
+      @current_user ||= find(id: session[:user_id])
```

Comment:
Isn't there a `current_user` method in ApplicationController?

First, this method belongs in ApplicationController. Secondly, if both exist, if will be confusing for the app, as to which method to call...

Reply:
Sorry, was so caught up in the whole null object pattern forgot I already had the method in application controller!